### PR TITLE
Upgrade vector

### PR DESCRIPTION
**Pull request by maltesander**
# Description

WIP - just for issue reference

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/templates/kuttl/logging/opa-vector-aggregator-values.yaml.j2
Comment (siegfriedweber) [inline tests/templates/kuttl/logging/opa-vector-aggregator-values.yaml.j2:1]: `filteredInvalidEvents` should be added and the `sinks` must be adapted:

```yaml
    filteredInvalidEvents:
      type: filter
      inputs: [vector]
      condition: |-
        .timestamp == from_unix_timestamp!(0) ||
        is_null(.level) ||
        is_null(.logger) ||
        is_null(.message)
  sinks:
    out:
      inputs: [filtered*]
{% if lookup('env', 'VECTOR_AGGREGATOR') %}
      type: vector
      address: {{ lookup('env', 'VECTOR_AGGREGATOR') }}
      buffer:
        # Avoid back pressure from VECTOR_AGGREGATOR. The test should
        # not fail if the aggregator is not available.
        when_full: drop_newest
{% else %}
      type: blackhole
{% endif %}
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: tests/templates/kuttl/logging/opa-vector-aggregator-values.yaml.j2
Comment (maltesander) [reply tests/templates/kuttl/logging/opa-vector-aggregator-values.yaml.j2:1]: https://github.com/stackabletech/opa-operator/pull/465/commits/83651a6d46d7400562fd6f66a371be0082f94fe5

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
LGTM but could you please also copy https://github.com/stackabletech/operator-templating/blob/main/template/tests/kuttl-test.yaml.jinja2 into this pull request?

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**